src/object/srv_obj.c

```diff
@@ -921,8 +926,11 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	rc = ds_pre_check(orw->orw_oid, orw->orw_map_ver, orw->orw_pool_uuid,
 			  orw->orw_co_hdl, orw->orw_co_uuid,
 			  opc_get(rpc->cr_opc), &cont_hdl, &cont);
-	if (rc)
-		goto out;
+	if (rc != 0) {
+		D_ASSERTF(rc < 0, "unexpected error# %d\n", rc);
```

Comment:
are we sure we never hit that case?
we don't want to assert servers if it's possible.

Reply:
ds_pre_check() is totally server side logic, whether it returns positive value is neither affected by client nor storage. So assert() here can guarantee that related server side logic is correct. That is safe.